src/group_theory/schur_zassenhaus.lean

```diff
@@ -162,21 +164,204 @@ begin
     apply_instance },
 end
 
-/-- **Schur-Zassenhaus** for abelian normal subgroups:
-  If `H : subgroup G` is abelian, normal, and has order coprime to its index, then there exists
-  a subgroup `K` which is a (right) complement of `H`. -/
-theorem exists_right_complement'_of_coprime [fintype G] [H.normal]
+/-- Do not use this lemma: It is made obsolete by `exists_right_complement'_of_coprime` -/
+private lemma exists_right_complement'_of_coprime_aux [fintype G] [H.normal]
   (hH : nat.coprime (fintype.card H) H.index) :
   ∃ K : subgroup G, is_complement' H K :=
 nonempty_of_inhabited.elim
   (λ α : H.quotient_diff, ⟨mul_action.stabilizer G α, is_complement'_stabilizer_of_coprime hH⟩)
 
-/-- **Schur-Zassenhaus** for abelian normal subgroups:
-  If `H : subgroup G` is abelian, normal, and has order coprime to its index, then there exists
-  a subgroup `K` which is a (left) complement of `H`. -/
-theorem exists_left_complement'_of_coprime [fintype G] [H.normal]
-  (hH : nat.coprime (fintype.card H) H.index) :
-  ∃ K : subgroup G, is_complement' K H :=
-Exists.imp (λ _, is_complement'.symm) (exists_right_complement'_of_coprime hH)
+end schur_zassenhaus_abelian
+
+open_locale classical
+
+universe u
+
+namespace schur_zassenhaus_induction
+
+variables {G : Type u} [group G] [fintype G] {N : subgroup G} [normal N]
+  (h1 : nat.coprime (fintype.card N) N.index)
+  (h2 : ∀ (G' : Type u) [group G'] [fintype G'], by exactI
+    ∀ (hG'3 : fintype.card G' < fintype.card G)
+    {N' : subgroup G'} [N'.normal] (hN : nat.coprime (fintype.card N') N'.index),
+    ∃ H' : subgroup G', is_complement' N' H')
+  (h3 : ∀ H : subgroup G, ¬ is_complement' N H)
+
+include h1 h2 h3
+
```

Comment:
Could you please include some comments that outline the 7-step strategy?

Reply:
Pushed a change.